x-pack/filebeat/input/entityanalytics/provider/activedirectory/activedirectory.go

```diff
@@ -157,16 +157,17 @@ func (p *adInput) Run(inputCtx v2.Context, store *kvstore.Store, client beat.Cli
 			}
 			updateTimer.Reset(p.cfg.UpdateInterval)
 			p.logger.Debugf("Next update expected at: %v", time.Now().Add(p.cfg.UpdateInterval))
-		case <-updateTimer.C:
-			start := time.Now()
-			if err := p.runIncrementalUpdate(inputCtx, store, client); err != nil {
+			last = start
+		case start := <-updateTimer.C:
+			if err := p.runIncrementalUpdate(inputCtx, store, last, client); err != nil {
 				p.logger.Errorw("Error running incremental update", "error", err)
 				p.metrics.updateError.Inc()
 			}
 			p.metrics.updateTotal.Inc()
 			p.metrics.updateProcessingTime.Update(time.Since(start).Nanoseconds())
 			updateTimer.Reset(p.cfg.UpdateInterval)
 			p.logger.Debugf("Next update expected at: %v", time.Now().Add(p.cfg.UpdateInterval))
+			last = start
```

Comment:
If time was used I think it would be better to get it from `state.whenChanged` in order to minimize issues from disagreements between client and server time (specifically, when the server doesn't provide all data up to the time of a request).

Reply:
It's a little more subtle than that, but I think I have got what you want in my next change.